### PR TITLE
Feat/water 2579 annual billing query

### PR DESCRIPTION
**Pull request by james-siteclick**
Modifies the annual billing query to avoid re-processing charge versions in a given financial year.
Moves code for creation of billing_batch_charge_versions to a new repo and fixes camel-casing.

### CODE REVIEW

File: src/lib/connectors/bookshelf/ChargeVersion.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+module.exports = bookshelf.model('ChargeVersion', {
+  tableName: 'charge_versions',
+
+  hasTimestamps: ['date_created', 'date_updated'],
+
+  idAttribute: 'charge_version_id',
+
+  chargeElements () {
+    return this.hasMany('ChargeElement', 'charge_version_id', 'charge_version_id');
```
Comment (Stephan-Hough) [inline src/lib/connectors/bookshelf/ChargeVersion.js]: 'charge_version_id' has been added twice.

File: src/lib/connectors/repos/queries/billing-batch-charge-versions.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  and l.region_id = :regionId::uuid
+  and (cv.end_date is null or cv.end_date > :fromDate)
+returning *;`;
+
+exports.createAnnual = `
+    insert into water.billing_batch_charge_versions (billing_batch_id, charge_version_id)
+    select :billingBatchId, cv.charge_version_id
+    from water.licences l
+      join water.charge_versions cv on l.licence_ref = cv.licence_ref
+      join water.charge_elements ce on cv.charge_version_id = ce.charge_version_id
```
Comment (Stephan-Hough) [inline src/lib/connectors/repos/queries/billing-batch-charge-versions.js]: Do you need to join the charge element table here? Is it used as a constraint?

File: src/lib/connectors/repos/queries/billing-batch-charge-versions.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      )
+    returning *;`;
+
+exports.createTwoPartTariff = `
+      insert into water.billing_batch_charge_versions (billing_batch_id, charge_version_id)
+      select :billingBatchId, cv.charge_version_id
+      from water.licences l
+        join water.charge_versions cv on l.licence_ref = cv.licence_ref
+        join water.licence_agreements la on l.licence_ref = la.licence_ref
+        join water.charge_elements ce on cv.charge_version_id = ce.charge_version_id
```
Comment (Stephan-Hough) [inline src/lib/connectors/repos/queries/billing-batch-charge-versions.js]: Same as above comment.

File: src/lib/connectors/bookshelf/ChargeVersion.js (hunk 1 above)
Comment (james-siteclick) [reply src/lib/connectors/bookshelf/ChargeVersion.js]: Once is for the local table and once for the foreign key - I think this is OK

File: src/lib/connectors/repos/queries/billing-batch-charge-versions.js (hunk 2 above)
Comment (james-siteclick) [reply src/lib/connectors/repos/queries/billing-batch-charge-versions.js]: It ensures any charge versions won't be processed if they don't contain any charge elements, so is quite handy from that respect

File: src/lib/connectors/repos/queries/billing-batch-charge-versions.js (hunk 3 above)
Comment (james-siteclick) [reply src/lib/connectors/repos/queries/billing-batch-charge-versions.js]: I think in this case it ensures we only process charge versions which have a charge element matching the season of the batch type (summer or winter/all year)

File: src/lib/connectors/repos/queries/billing-batch-charge-versions.js (hunk 2 above)
Comment (james-siteclick) [reply src/lib/connectors/repos/queries/billing-batch-charge-versions.js]: I think here it ensures we only process charge versions which have charge elements

**Pull request merged**